### DISCUSSION TITLE
Searching for repositories that use a specific file extension

### DISCUSSION THREAD

**Discussion by ljsikuade** [community/community#24490] [Apps, API and Webhooks]
<p>Couldn’t see this in the dropdown. I’m trying to phrase a query with the Search API that’ll return a list of all public repositories that have a specific file extension. </p>
<p>For example if I wanted to get all that contain a .json file. </p>
<p>Right now I’ve tried: </p>
<p><a href="https://api.github.com/search/extension:.%5Bextensionname%5D&amp;type=Code" rel="nofollow noopener">https://api.github.com/search/extension:.[extensionname]&amp;type=Code</a><br>
But I’m struggling to find good examples I can use.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hey <a class="mention" href="/ljsikuade">@ljsikuade</a>,</p>
<p>Thanks for being here! You can search by file extension, per the docs:</p>
<p>Suppose you want to search for popular Tetris repositories written in Assembly. Your query might look like this.</p>
<pre><code>curl https://api.github.com/search/repositories?q=tetris+language:assembly&amp;sort=stars&amp;order=desc
</code></pre>
<p>You can search for multiple topics by adding more <code>topic:</code> instances, and including the <code>mercy-preview</code> header. For example:</p>
<pre><code>curl -H "Accept: application/vnd.github.mercy-preview+json" \
https://api.github.com/search/repositories?q=topic:ruby+topic:rails
</code></pre>
<p>In this request, we’re searching for repositories with the word <code>tetris</code> in the name, the description, or the README. We’re limiting the results to only find repositories where the primary language is Assembly. </p>
<p>Check out the <a href="https://developer.github.com/v3/search/#example" rel="nofollow noopener">docs</a> for more info.</p>
<p>I hope this helps!</p>
<p>Andrea</p>

**Comment by 8SquirrelCOde8** [comment]
<p><a href="https://api.github.com/search/extension:.%5Bextensionname%5D&amp;type=Code" rel="nofollow noopener">https://api.github.com/search/extension:.[extensionname]&amp;type=Code</a></p>
